tune-tikv-thread-performance.md

```diff
@@ -0,0 +1,86 @@
+---
+title: Tune TiKV Thread Pool Performance
+summary: Learn how to tune TiKV thread pools for optimal performance.
+category: tuning
+---
+
+# Tune TiKV Thread Pool Performance
+
+This document introduces TiKV internal thread pools and how to tune their performance.
+
+## Thread pool introduction
+
+In TiKV 4.0, the TiKV thread pool is mainly composed of gRPC, Scheduler, UnifyReadPool, Raftstore, Apply, RocksDB, and some scheduled tasks and detection components that do not consume much CPU. This document mainly introduces a few CPU-intensive thread pools that affect the performance of read and write requests.
+
+* The gRPC thread pool: it handles all network requests and forwards requests of different task types to different thread pools.
+
+* The Scheduler thread pool: it is responsible for detecting write transaction conflicts, transforming requests like the two-phase commit, pessimistic locking, and transaction rollbacks into key-value pair arrays, and then giving them to the Raftstore thread for Raft log replication.
+
+* The Raftstore thread pool: it is responsible for processing all Raft messages and the proposal to add a new log, and writing the log to a disk. When the logs in the majority of replicas are consistent, this thread pool will send the log to the Apply thread.
+
+* The Apply thread pool: it receives the submitted log sent from the Raftstore thread pool, parses it as a key-value request, then writes it to RocksDB, calls the callback function to notify the gRPC thread pool that the write request is complete, and returns the result to the client.
+
+* The RocksDB thread pool: it is a thread pool for RocksDB to compact and flush tasks. For RocksDB's architecture and `Compact` operation, refer to [RocksDB: A Persistent Key-Value Store for Flash and RAM Storage](https://github.com/facebook/rocksdb).
+
+* The UnifyReadPool thread pool: it is a new feature introduced in TiKV 4.0. It comes from the merging of the previous Coprocessor thread pool and Storage Read Pool. All read requests such as kv get, kv batch get, raw kv get, coprocessor are executed in this thread pool.
+
+## TiKV read-only requests
+
+TiKV's read requests are divided into the following two types:
```

Comment:
```suggestion
TiKV's read requests are divided into the following types:
```
Suggest not specifying the number here to avoid further edits due to changes to the following list.